### PR TITLE
Easy 2028  allow nested zips

### PR DESCRIPTION
**Pull request by stephanHogenboom**
Fixes Easy-2028  allow nested zips

#### When applied it will
* posting nested zips as datasets 

- [x]  test on deasy
- [x] update specs?

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-deposit-api                      | [PR#129](https://github.com/DANS-KNAW/easy-deposit-api/pull/129)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala [hunk 1]
```diff
@@ -68,18 +70,14 @@ package object servlets extends DebugEnhancedLogging {
 
   implicit class RichZipInputStream(val zipInputStream: ZipInputStream) extends AnyVal {
     def unzipPlainEntriesTo(dir: File): Try[Unit] = {
-      def extract(entry: ZipEntry) = {
+      def extract(entry: ZipEntry): Try[Any] = {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]:
```suggestion
      def extract(entry: ZipEntry): Try[Unit] = {
```

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala [hunk 2]
```diff
@@ -68,18 +70,14 @@ package object servlets extends DebugEnhancedLogging {
...
-            Failure(BadRequestException(s"ZIP file is malformed. It contains a Zip ${ entry.getName }."))
-          else {
-            logger.info(s"Extracting ${ entry.getName } size=${ entry.getSize } compressedSize=${ entry.getCompressedSize } CRC=${ entry.getCrc }")
-            Try(Files.copy(zipInputStream, (dir / entry.getName).path))
-          }.recoverWith {
-            case e if e.isInstanceOf[ZipException] => Failure(BadRequestException(s"ZIP file is malformed. $e"))
-          }
+          logger.info(s"Extracting ${ entry.getName } size=${ entry.getSize } compressedSize=${ entry.getCompressedSize } CRC=${ entry.getCrc }")
+          Try(Files.copy(zipInputStream, (dir / entry.getName).path))
+        }.recoverWith {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: is this `recoverWith` applicable to the `else` branch only, or also to the `if` branch? That isn't clear to me from the code. Not sure if you have the right functionality now or if you're introducing a bug in error handling.

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala [hunk 3]
```diff
@@ -205,6 +204,61 @@ class UploadSpec extends DepositServletFixture {
...
+    val bagDir = testDir / "drafts" / "foo" / uuid.toString / "bag"
+    val absoluteTarget = (bagDir / "data" / relativeTarget).createDirectories()
+    absoluteTarget.list.size shouldBe 0 // precondition
+    post(
+      uri = s"/deposit/$uuid/file/$relativeTarget",
+      params = Iterable(),
+      headers = Seq(fooBarBasicAuthHeader),
+      files = Seq(("formFieldName", (testDir / "input/2.zip").toJava))
+    ) {
+      body shouldBe ""
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala]:
```suggestion
      body shouldBe empty
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala [hunk 4]
```diff
@@ -205,6 +204,61 @@ class UploadSpec extends DepositServletFixture {
...
+    val uuid = createDeposit
+    val relativeTarget = "path/to/dir"
+    val bagDir = testDir / "drafts" / "foo" / uuid.toString / "bag"
+    val absoluteTarget = (bagDir / "data" / relativeTarget).createDirectories()
+    absoluteTarget.list.size shouldBe 0 // precondition
+    post(
+      uri = s"/deposit/$uuid/file/$relativeTarget",
+      params = Iterable(),
+      headers = Seq(fooBarBasicAuthHeader),
+      files = Seq(("formFieldName", (testDir / "input/2.zip").toJava))
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala]: just for readability:

```suggestion
      files = Seq("formFieldName" -> (testDir / "input/2.zip").toJava)
```

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala (hunk 1 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: Does not compile, the if returns a Try[File], the else a Try[Long]

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala (hunk 2 above)
Comment (stephanHogenboom) [reply src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: I guess only to the else clause. But i assume 
`Try((dir / entry.getName).createDirectories())` is not able to cause a ZipException

**Pull request merged**